security-advisories/README.md

```diff
@@ -14,9 +14,178 @@ At this page we will list of all known security vulnerabilities found on OP-TEE.
+disclosure.
+
+**optee_os.git:**
+ - [core: svc: always check ta parameters (d5c5b0b77b2)](
+https://github.com/OP-TEE/optee_os/commit/d5c5b0b77b2b589666024d219a8007b3f5b6faeb)
+
+| Reported by  | CVE ID    | OP-TEE ID        | Affected versions  |
+| ------------ | :-------: | :--------------: | ------------------ |
+| [Riscure]    | Not/Ready | OP-TEE-2018-0007 | v3.3.0 and earlier |
+
+### Potential disclosure of previously loaded TA code and data
+The function `elf_load_body` is used to load the code and data segments while
+dynamically loading a TA. The amount of memory allocated for the code and data
+segments is previously determined and the sum of it is stored in
+`state->vasize`. The actual allocated amount of memory is rounded up the next
+multiple of the memory pool granularity. To ensure that the newly loaded TA is
+not able to observe any data belonging to a TA previously stored on this exact
+location in memory, the memory block is set to zero. The size used to memset
+the block to zero is the sum of the sizes of the segments, not the rounded size
+of the actual allocation. This means that the remaining space at the end of the
+allocation is not cleared, potentially leaking code and/or data of a previous
+TA. The information gained by this attack is limited by the memory layout of
+the (compromised) TA performing the attack and the flags (i.e. is unloading the
+TA prevented when the last session is closed due to
+`TA_FLAG_INSTANCE_KEEP_ALIVE`) and layout of the attacked TA. Therefore we
+consider the risk to be
+
+**optee_os.git:**
+ - [core: clear the entire TA area (7e768f8a473)](
+https://github.com/OP-TEE/optee_os/commit/7e768f8a473409215fe3fff8f6e31f8a3a0103c6)
+
+| Reported by  | CVE ID    | OP-TEE ID        | Affected versions  |
+| ------------ | :-------: | :--------------: | ------------------ |
+| [Riscure]    | Not/Ready | OP-TEE-2018-0006 | v3.3.0 and earlier |
+
+### tee_mmu_check_access_rights does not check final page of TA buffer
+The function `tee_mmu_check_access_rights` is used to check access rights to a
+given memory region. This function is used when a TA performs a system call to
+verify that the TA has the correct access rights to the buffer it provides.
+However, the function `tee_mmu_check_access_rights` does not check every page of
+the TA provided buffer. A TA provides a buffer as a pointer (uaddr) and a
+length (a). The provided buffer is checked piecewise in increments of addr_incr
+(4KiB) in a for-loop. In the case where len is not already page aligned, the
+termination condition a < (uaddr + len) has been passed when addr_incr is added
+the last time of the loop iteration. Therefore, the final page of the TA
+provided buffer is not checked. A TA could provide a buffer of which up to 4KiB
+resides in the context of the TEE kernel or another TA. This could lead to
+memory corruption of the TEE itself or another TA. Memory corruption
+vulnerabilities can have
```

Comment:
...can have what?

Reply:
Copy/paste bug from the Riscure doc, I'll fix on Monday.